vstest build failing silently in source-build

Restore is failing for datacollector.csproj when building with source build:

```
Package: Publish /src/bin/src/vstest.1e37ee32dbba48ad797b03bff0058c5c60a3a770/src/datacollector/datacollector.csproj[0m
Microsoft (R) Build Engine version 16.1.76+g14b0a930a7 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 51.73 ms for /src/bin/src/vstest.1e37ee32dbba48ad797b03bff0058c5c60a3a770/src/Microsoft.TestPlatform.CommunicationUtilities/Microsoft.TestPlatform.CommunicationUtilities.csproj.
  Restore completed in 51.73 ms for /src/bin/src/vstest.1e37ee32dbba48ad797b03bff0058c5c60a3a770/src/Microsoft.TestPlatform.Common/Microsoft.TestPlatform.Common.csproj.
  Restore completed in 0.76 ms for /src/bin/src/vstest.1e37ee32dbba48ad797b03bff0058c5c60a3a770/src/Microsoft.TestPlatform.CoreUtilities/Microsoft.TestPlatform.CoreUtilities.csproj.
  Restore completed in 1.23 ms for /src/bin/src/vstest.1e37ee32dbba48ad797b03bff0058c5c60a3a770/src/Microsoft.TestPlatform.CrossPlatEngine/Microsoft.TestPlatform.CrossPlatEngine.csproj.
  Restore completed in 0.87 ms for /src/bin/src/vstest.1e37ee32dbba48ad797b03bff0058c5c60a3a770/src/Microsoft.TestPlatform.ObjectModel/Microsoft.TestPlatform.ObjectModel.csproj.
  Restore completed in 0.8 ms for /src/bin/src/vstest.1e37ee32dbba48ad797b03bff0058c5c60a3a770/src/Microsoft.TestPlatform.PlatformAbstractions/Microsoft.TestPlatform.PlatformAbstractions.csproj.
  Restore completed in 1.37 ms for /src/bin/src/vstest.1e37ee32dbba48ad797b03bff0058c5c60a3a770/src/Microsoft.TestPlatform.Utilities/Microsoft.TestPlatform.Utilities.csproj.
/src/bin/src/vstest.1e37ee32dbba48ad797b03bff0058c5c60a3a770/src/datacollector/datacollector.csproj : error NU1201: Project Microsoft.TestPlatform.CommunicationUtilities is not compatible with net451 (.NETFramework,Version=v4.5.1). Project Microsoft.TestPlatform.CommunicationUtilities supports: netstandard2.0 (.NETStandard,Version=v2.0)
/src/bin/src/vstest.1e37ee32dbba48ad797b03bff0058c5c60a3a770/src/datacollector/datacollector.csproj : error NU1201: Project Microsoft.TestPlatform.ObjectModel is not compatible with net451 (.NETFramework,Version=v4.5.1). Project Microsoft.TestPlatform.ObjectModel supports: netstandard2.0 (.NETStandard,Version=v2.0)
/src/bin/src/vstest.1e37ee32dbba48ad797b03bff0058c5c60a3a770/src/datacollector/datacollector.csproj : error NU1201: Project Microsoft.TestPlatform.CrossPlatEngine is not compatible with net451 (.NETFramework,Version=v4.5.1). Project Microsoft.TestPlatform.CrossPlatEngine supports: netstandard2.0 (.NETStandard,Version=v2.0)
/src/bin/src/vstest.1e37ee32dbba48ad797b03bff0058c5c60a3a770/src/datacollector/datacollector.csproj : error NU1201: Project Microsoft.TestPlatform.CoreUtilities is not compatible with net451 (.NETFramework,Version=v4.5.1). Project Microsoft.TestPlatform.CoreUtilities supports: netstandard2.0 (.NETStandard,Version=v2.0)
/src/bin/src/vstest.1e37ee32dbba48ad797b03bff0058c5c60a3a770/src/datacollector/datacollector.csproj : error NU1201: Project Microsoft.VisualStudio.TestPlatform.Common is not compatible with net451 (.NETFramework,Version=v4.5.1). Project Microsoft.VisualStudio.TestPlatform.Common supports: netstandard2.0 (.NETStandard,Version=v2.0)
/src/bin/src/vstest.1e37ee32dbba48ad797b03bff0058c5c60a3a770/src/datacollector/datacollector.csproj : error NU1201: Project Microsoft.TestPlatform.CommunicationUtilities is not compatible with net451 (.NETFramework,Version=v4.5.1) / win7-x64. Project Microsoft.TestPlatform.CommunicationUtilities supports: netstandard2.0 (.NETStandard,Version=v2.0)
/src/bin/src/vstest.1e37ee32dbba48ad797b03bff0058c5c60a3a770/src/datacollector/datacollector.csproj : error NU1201: Project Microsoft.TestPlatform.ObjectModel is not compatible with net451 (.NETFramework,Version=v4.5.1) / win7-x64. Project Microsoft.TestPlatform.ObjectModel supports: netstandard2.0 (.NETStandard,Version=v2.0)
/src/bin/src/vstest.1e37ee32dbba48ad797b03bff0058c5c60a3a770/src/datacollector/datacollector.csproj : error NU1201: Project Microsoft.TestPlatform.CrossPlatEngine is not compatible with net451 (.NETFramework,Version=v4.5.1) / win7-x64. Project Microsoft.TestPlatform.CrossPlatEngine supports: netstandard2.0 (.NETStandard,Version=v2.0)
/src/bin/src/vstest.1e37ee32dbba48ad797b03bff0058c5c60a3a770/src/datacollector/datacollector.csproj : error NU1201: Project Microsoft.TestPlatform.CoreUtilities is not compatible with net451 (.NETFramework,Version=v4.5.1) / win7-x64. Project Microsoft.TestPlatform.CoreUtilities supports: netstandard2.0 (.NETStandard,Version=v2.0)
/src/bin/src/vstest.1e37ee32dbba48ad797b03bff0058c5c60a3a770/src/datacollector/datacollector.csproj : error NU1201: Project Microsoft.VisualStudio.TestPlatform.Common is not compatible with net451 (.NETFramework,Version=v4.5.1) / win7-x64. Project Microsoft.VisualStudio.TestPlatform.Common supports: netstandard2.0 (.NETStandard,Version=v2.0)
  Restore failed in 394.67 ms for /src/bin/src/vstest.1e37ee32dbba48ad797b03bff0058c5c60a3a770/src/datacollector/datacollector.csproj.
cp: cannot stat ‘src/testhost/bin/Release/net451/win7-x64/*’: No such file or directory
cp: cannot stat ‘src/testhost.x86/bin/Release/net451/win7-x86/*’: No such file or directory
```

--- a/patches/vstest/0003-Add-the-condition-for-Source-build-2144.patch
+++ b/patches/vstest/0003-Add-the-condition-for-Source-build-2144.patch
@@ -1,0 +1,24 @@
+From 1d5e62d0d67a214806827d93049df2bdd126aa39 Mon Sep 17 00:00:00 2001
+From: Sarabjot Singh <sasin@microsoft.com>
+Date: Tue, 27 Aug 2019 17:35:20 +0530
+Subject: [PATCH] Add the condition for Source build (#2144)
+
+---
+ src/datacollector/datacollector.csproj | 1 +
+ 1 file changed, 1 insertion(+)
+
+diff --git a/src/datacollector/datacollector.csproj b/src/datacollector/datacollector.csproj
+index ef76109..2aa91a6 100644
+--- a/src/datacollector/datacollector.csproj
++++ b/src/datacollector/datacollector.csproj
+@@ -10,6 +10,7 @@
+   <PropertyGroup>
+     <AssemblyName>datacollector</AssemblyName>
+     <TargetFrameworks>netcoreapp2.1;net451</TargetFrameworks>
++    <TargetFrameworks Condition=" '$(DotNetBuildFromSource)' == 'true' ">netcoreapp2.1</TargetFrameworks>
+     <WarningsAsErrors>true</WarningsAsErrors>
+     <PlatformTarget Condition="'$(TargetFramework)' == 'net451'">AnyCPU</PlatformTarget>
+     <OutputType>Exe</OutputType>
+-- 
+1.8.3.1
+

